php/git.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+class Git {
+
+    /**
+     * Retrieve the commit ID of the current HEAD.
+     * @return string|null commit ID or null on failure
+     */
+    public static function getCurrentCommitID() {
+        $head = file_get_contents(".git/HEAD");
```

Comment:
`file_get_contents` can return false, but `preg_match` doesn't seem to care (at least not atm).

Reply:
FWIW, that MediaWiki class does the same thing

https://github.com/wikimedia/mediawiki/blob/master/includes/GitInfo.php#L215

Reply:
Yeah. My comment wasn't necessarily anything to fix; more me reviewing the code and leaving notes for myself etc too